docs/migrating-from-knative-build.md

```diff
@@ -0,0 +1,143 @@
+# Migrating from [Knative Build](https://github.com/knative/build)
+
+This doc describes a process for users who are familiar with Knative `Build` and
+`BuildTemplate` resources to migrate to Tekton `TaskRuns` and `Tasks`, respectively.
+
+Tekton's resources are heavily influenced by Knative's Build-related resources,
+with some additional features that enable them to be chained together inside a
+Pipeline, and provide additional flexibility and reusability.
+
+| **Knative**          | **Tekton**  |
+|----------------------|-------------|
+| Build                | TaskRun     |
+| BuildTemplate        | Task        |
+| ClusterBuildTemplate | ClusterTask |
+
+## Important differences
+
+* All `steps` must have a `name`. Step statuses are not reported in a
+  TaskRun's `status` in the same order they're specified in the `spec`, so the
+  `name` is used to associate the `status` with the step.
+
+* BuildTemplaate `parameters` are moved inside Task's `input.params` field, and
+  parameter placeholder strings (e.g., `${FOO}`) must be specified like
+  `${input.parameters.FOO}`.
+
+* Tasks must specify `input.resources` if they need to operate on a resource
+  (e.g., source from a Git repo). BuildTemplates did not specify input
+  resource requirements, and just assumed whatever source was available.
```

Comment:
can link to https://github.com/tektoncd/pipeline/blob/master/docs/tasks.md#input-resources